Frameworks/Foundation/NSNumberFormatter.mm

```diff
@@ -278,25 +279,17 @@ - (NSString*)currencySymbol {
  @Status Interoperable
 */
 - (void)setNilSymbol:(NSString*)symbol {
-    @synchronized(self) {
-        if (self.nilSymbol != symbol) {
-            [self.nilSymbol autorelease];
-            self.nilSymbol = [symbol copy];
-        }
-    }
+    _nilSymbol = symbol;
```

Comment:
This should probably be `_nilSymbol.attach([symbol copy])` to uphold the property contract.